### PR TITLE
fix unauthentication logic and acces token gen

### PR DESCRIPTION
**Pull request by dbalucas**

### CODE REVIEW

File: src/lennybot/check/docker_image_available.py [hunk 1]
```diff
@@ -114,20 +114,25 @@ def _authenticate_on_registry(self, registry: str, authentication_header: WwwAut
...
+
             access_token = token_data.get("token")
+
+            if token_data.get("token") is None:
+                access_token = token_data.get("access_token")
+
             return str(access_token)
 
         if response.status_code == 401:
+            logging.debug("error during authentication:", response.status_code, response.headers)
```
Comment (raynigon) [inline src/lennybot/check/docker_image_available.py:135]:
```suggestion
            logging.error("Authentication failed:", response.status_code, response.headers)
```

File: src/lennybot/check/docker_image_available.py [hunk 2]
```diff
@@ -114,20 +114,25 @@ def _authenticate_on_registry(self, registry: str, authentication_header: WwwAut
             "access_type": "offline",
         }
 
-        if registry in self._container_config.registries.keys():
-            registry_data = self._container_config.registries[registry]
-            params["password"] = registry_data.password
-            params["username"] = registry_data.username
+        registry_data = self._container_config.registries[registry]
+        password = registry_data.password
+        username = registry_data.username
```
Comment (raynigon) [inline src/lennybot/check/docker_image_available.py:119]: Whats happens if the registry does not exist?

**Review by raynigon — CHANGES_REQUESTED** [review]
Please fix

File: src/lennybot/check/docker_image_available.py (hunk 2 above)
Comment (dbalucas) [reply src/lennybot/check/docker_image_available.py]: then it ends up in this if:

`if registry not in self._container_config.registries.keys():
            logging.debug("Registry not found in config")
            response = requests.get(url)`

**Review by raynigon — APPROVED** [review]
LGTM

**Pull request merged**